crates/api/src/wire.proto

```diff
@@ -0,0 +1,61 @@
+syntax = "proto3";
+
+package kitsune2.wire;
+
+// A Kitsune2 wire protocol message.
+//
+// This is the top-level encoding
+// that will be transferred between Kitsune2 peers. Most communications
+// between peers to make Kitsune2 actually function will be encoded
+// separately inside the payload of TY_MODULE type messages.
+message K2Proto {
+  // Enumeration of the types of messages that can be sent between peers.
+  //
+  // We are using this enum field to distinguish between top-level messages,
+  // rather than protobuf's oneof because of the downsides of upgrading oneofs.
+  enum Ty {
+    // The "UNSPECIFIED" type for future message types.
+    //
+    // In general, peers should ignore unspecified messages, but
+    // should still count them toward any ratelimiting metrics.
+    TY_UNSPECIFIED = 0;
+
+    // This message is preflight data.
+    //
+    // The implementor is responsible for encoding any module-specific
+    // requirements within the data payload of this message type.
+    // For example, if peers are required to include the "dht_v1" module
+    // to communicate with each other, they should reject preflight to
+    // peers that do not include that module.
+    TY_PREFLIGHT = 1;
+
+    // This is a notification or fire-and-forget message from a peer.
+    //
+    // This type requires that a "space" be specified.
+    TY_NOTIFY = 2;
+
+    // This is a module communication.
+    //
+    // This type requires that a "space" be specified.
```

Comment:
Is this line a copy-paste error or does this type require both?

Reply:
If I'm understanding this then yes both are required. So you might send
- abc, gossip, please give me your new ops since we last spoke
- abc, fetch, please give me the op data for this list of op hashes

Or maybe we won't ask so nicely, but I think that's the idea here